IZP2tester_zs2020.py

```diff
@@ -18,87 +19,87 @@
 cbLGRAY = '\033[1;37m'
 cbCYAN = '\033[1;46m'
 
+class TestResult:
+    SEGFAULT = "segfault"
+    ERROR = "error"
+    BINARY = "binary"
+    TIMEOUT = "timeout"
+    OK = "ok"
+    MEM_ERROR = "memerror"
+
+    def __init__(self, type, msg=''):
+        self.type = type
+        self.msg = msg
+
+def _run_test(executable, args, filename, memcheck=False, maxstack=False):
+    tmp_file_name = filename + '.tmp'
+    arguments = (['valgrind', '--log-fd=1', '-q', '--leak-check=full'] if memcheck else []) + \
+                (['--max-stackframe=4040064'] if memcheck and maxstack else []) + \
+                [f'./{executable}'] + args + [tmp_file_name]
+
+    shutil.copyfile(filename, tmp_file_name)
+
+    ret_type = None
+    ret_msg = None
+    try:
+        p = subprocess.Popen(arguments, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = p.communicate(timeout=3)
+
+        if p.returncode:
+            raise subprocess.CalledProcessError(p.returncode - 256, ' '.join(arguments))
+        if out:
```

Comment:
How well does this handle programs with debug stdout output?

Reply:
Well, I hope everyone is printing to `stderr`. I could redirect valgrind output to a different fd instead, but that'd require some shell fd redirections and I don't think I'm able to do that without making mistakes.